examples/saas-clients/amazons3/working-with-amazons3-client/README.md

```diff
@@ -0,0 +1,422 @@
+    resource function deleteObject(http:Caller caller, http:Request request, string bucketName, string objectName) {
+        // Create AmazonS3 client with the above amazonS3Config.
+        amazons3:AmazonS3Client|error amazonS3Client = new(amazonS3Config);
+        if (amazonS3Client is amazons3:AmazonS3Client) {
+            // Define new response. 
+            http:Response backendResponse = new();
+            error? response = amazonS3Client->deleteObject(untaint bucketName, untaint objectName);
+            if (response is error) {
+                // Send the error response.
+                createAndSendErrorResponse(caller, untaint <string>response.detail().message, 
+                                        OBJECT_DELETION_ERROR_MSG);
+            } else {
+                // If there is no error, then object deleted successfully. Send the success response.
+                backendResponse.setTextPayload(untaint string `${objectName} deleted from Amazon S3 bucket : ${bucketName}.`, 
+                                        contentType = "text/plain");
+                respondAndHandleError(caller, backendResponse, RESPOND_ERROR_MSG);
+            }
+        } else {
+            // Send the error response.
+            createAndSendErrorResponse(caller, <string>amazonS3Client.detail().message, CLIENT_CREATION_ERROR_MSG);
+        }
+    }
+
+    @http:ResourceConfig {
+        methods: ["DELETE"],
+        path: "/{bucketName}"
+    }
+    // Function to delete bucket.
+    resource function deleteBucket(http:Caller caller, http:Request request, string bucketName, string objectName) {
+        // Create AmazonS3 client with the above amazonS3Config.
+        amazons3:AmazonS3Client|error amazonS3Client = new(amazonS3Config);
+        if (amazonS3Client is amazons3:AmazonS3Client) {
+            // Define new response. 
+            http:Response backendResponse = new();
+            // Invoke deleteBucket remote function from amazonS3Client.
+            error? response = amazonS3Client->deleteBucket(untaint bucketName);
+            if (response is error) {
+                // Send the error response.
+                createAndSendErrorResponse(caller, untaint <string>response.detail().message, 
+                                        BUCKET_DELETION_ERROR_MSG);
+            } else {
+                // If there is no error, then bucket deleted successfully. Send the success response.
+                backendResponse.setTextPayload(untaint string `${bucketName} deleted from Amazon S3.`, 
+                                        contentType = "text/plain");
+                respondAndHandleError(caller, backendResponse, RESPOND_ERROR_MSG);
+            }
+        } else {
+            // Send the error response.
+            createAndSendErrorResponse(caller, <string>amazonS3Client.detail().message, CLIENT_CREATION_ERROR_MSG);
+        }
+    }
+}
+
+// Function to extract the object content from request payload
```

Comment:
Add more inline comments to this section. Users need to know what you are doing in each line.